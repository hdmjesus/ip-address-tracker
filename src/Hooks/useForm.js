import { useState } from "react";

export const useForm = (initialState = {}) => {
  const [values, setValues] = useState(initialState);

  const reset = () => {
    setValues(initialState);
  };
  const handleInputChange = ({ target }) => {
    setValues({
      ...values,
      [target.name]: target.value,
    });
  };

  return [values, handleInputChange, reset];
};

// Este Hook nos permite obtener los valores y propiedades de un Input, ademas de tener los valores nuevos cuando este cambie.
