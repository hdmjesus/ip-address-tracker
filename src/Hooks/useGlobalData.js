import { useContext } from "react";
import { TextContext } from "../Context/StaticContext";

export function useGlobalData() {
  const data = useContext(TextContext);
  return data;
}
