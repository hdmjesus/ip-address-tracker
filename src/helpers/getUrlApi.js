import { apiKey, url } from "../config";

export const getUrlApi = (inputValue) => {
  let UrlApiIp;
  // Esta dondicion , devuelve el endpoint correspondiente a si es una direccion ip o un nombre de dominio
  if (
    inputValue.match(
      /^((25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}(25[0-5]|2[0-4]\d|[01]?\d\d?)$/
    )
  ) {
    UrlApiIp = `${url}?apiKey=${apiKey}&ipAddress=${inputValue}`;
    return UrlApiIp;
  } else {
    UrlApiIp = `${url}?apiKey=${apiKey}&domain=${inputValue}`;
    return UrlApiIp;
  }
};
