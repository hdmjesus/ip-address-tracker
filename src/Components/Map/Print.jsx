import React, { useEffect, useState } from "react";
import L from "leaflet";
import { useGlobalData } from "../../Hooks/useGlobalData";
import { Map, TileLayer, Marker } from "react-leaflet";
import "leaflet/src/core/Class";

export const MapPrint = () => {
  const { data, loading, error } = useGlobalData();
  const { location } = data || {};
  const { lat, lng } = location || {};

  // useEffect(() => {
  //   const { lat, lng } = location || {};
  //   if (lat) {
  //     let mymap = L.map("mapid").setView([lat, lng], 13);
  //     var marker = L.marker([lat, lng]).addTo(mymap);
  //     L.tileLayer("https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png?{foo}", {
  //       foo: "bar",
  //       attribution:
  //         '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors',
  //     }).addTo(mymap);
  //     return () => {
  //       if (lat) {
  //         L.remove(mymap);
  //       }
  //     };
  //   }
  // }, [location]);
  // center={(`${lat}`, `${lng}`)}
  // return <div id="mapid" className="z-0"></div>;

  return (
    <div>
      {location && (
        <Map center={[lat, lng]} zoom={13} id="mapid">
          <TileLayer
            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
            attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
          />
          <Marker position={[lat, lng]}></Marker>
        </Map>
      )}
    </div>
  );
};
