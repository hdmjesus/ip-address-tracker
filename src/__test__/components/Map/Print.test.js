import React from "react";
import { mount } from "enzyme";
import { renderHook, act } from "@testing-library/react-hooks";
import { useFetch } from "../../../Hooks/useFetch";
import { MapPrint } from "../../../Components/Map/Print";
import { getUrlApi } from "../../../helpers/getUrlApi";
import { Map, TileLayer, Marker } from "react-leaflet";

import { TextContext } from "../../../Context/StaticContext";
describe("pruebas en <Print/>", () => {
  const setValueInput = jest.fn();
  const { Provider } = TextContext;
  const { result, waitForNextUpdate } = renderHook(() =>
    useFetch(getUrlApi("8.8.8.8"))
  );

  test("debe mostrarse correctamente", async () => {
    await waitForNextUpdate({ timeout: 5000 });
    const { data, loading, error } = result.current;
    const wrapper = mount(
      <Provider value={{ data, loading, setValueInput }}>
        <MapPrint />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("Map").exists()).toBe(true);
    // const formSubmit = wrapper.find("form").prop("onSubmit");
  });
  test("debe mostrarse el marcador correctamente", async () => {
    const { data, loading, error } = result.current;

    const wrapper = mount(
      <Provider value={{ data, loading, setValueInput }}>
        <MapPrint />
      </Provider>
    );

    expect(wrapper.find("Marker").exists()).toBe(true);
    // const formSubmit = wrapper.find("form").prop("onSubmit");
  });
});
