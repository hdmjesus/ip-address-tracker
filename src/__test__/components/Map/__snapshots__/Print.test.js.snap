// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <Print/> debe mostrarse correctamente 1`] = `
<div>
  <div
    id="mapid"
  />
</div>
`;
